### PR TITLE
Proxy support

### PR DESCRIPTION
**Pull request by bgurney**
We've had to fork this package to add support for a proxy. If you think this would be useful to merge in then feel free. Any questions or suggestions just ping them over!

### CODE REVIEW

File: src/LiveChat/Api/Rest/RestRequest.php [hunk 1]
```diff
@@ -62,6 +64,16 @@ public function setPassword($password)
     }
 
     /**
+     * Set proxy
+     * @param string $proxy Url:Port
+     */
+    public function setProxy($proxy)
+    {
+        $this->proxyPort = parse_url( $proxy, PHP_URL_PORT ) ?: 80;
+        $this->proxy = str_replace( ':' . $this->proxyPort, '', $proxy );
```
Comment (HaGii) [inline src/LiveChat/Api/Rest/RestRequest.php]: Code formatting: please remove empty spaces before first and after last functions arguments.

File: src/LiveChat/Api/Rest/RestRequest.php [hunk 2]
```diff
@@ -252,6 +264,12 @@ private function setCurlOpts(&$curlHandle)
         curl_setopt($curlHandle, CURLOPT_URL, $this->url);
         curl_setopt($curlHandle, CURLOPT_RETURNTRANSFER, true);
         curl_setopt($curlHandle, CURLOPT_HTTPHEADER, $headers);
+
+        if ($this->proxy && $this->proxyPort)
+        {
```
Comment (HaGii) [inline src/LiveChat/Api/Rest/RestRequest.php]: Code formatting: please use `if (condition) {` structure. Note: there is no new line between `)` and `{` signs.

**Comment by HaGii** [comment]
Hi @bgurney, thanks a lot for commiting! I've noticed 2 code style issues. Please fix it and I'll merge your changes. 
Thanks!

**Comment by bgurney** [comment]
@HaGii I have committed these changes. Hope someone finds it useful!

**Comment by HaGii** [comment]
@bgurney Thanks a lot! :+1:

**Pull request merged**